gateway/ln-gateway/src/lib.rs

```diff
@@ -1433,10 +1462,39 @@ impl Gateway {
             )))
     }
 
+    async fn load_or_generate_mnemonic(&self) -> Result<Mnemonic> {
+        Ok(
+            if let Ok(entropy) =
+                Client::load_decodable_client_secret::<Vec<u8>>(&self.gateway_db).await
+            {
+                Mnemonic::from_entropy(&entropy)
+                    .map_err(|e| GatewayError::ClientCreationError(e.to_string()))?
+            } else {
+                let mnemonic = if let Ok(words) = std::env::var(FM_GATEWAY_MNEMONIC_ENV) {
```

Comment:
Currently the only way to recover from a mnemonic is to delete the database or recover using a new gateway with a different data directory. Is that ok from a user experience perspective? Thoughts?

Reply:
This seems fine for now, recovery in separate PR

Reply:
In a follow-up PR it would be nice to make setting the mnemonic an API call.